source/vir/src/ast_simplify.rs

```diff
@@ -354,7 +355,12 @@ fn simplify_one_expr(ctx: &GlobalCtx, state: &mut State, expr: &Expr) -> Result<
             }
             let mut conjunction: Expr = es[0].clone();
             for i in 0..ops.len() {
-                let op = BinaryOp::Inequality(ops[i]);
+                let op: BinaryOp;
+                if let ChainedOp::CmpOp(a) = ops[i] {
+                    op = BinaryOp::Inequality(a);
+                } else {
+                    op = BinaryOp::Eq(Mode::Spec);
+                }
```

Comment:
This can be written as `let op = match ops[i] { ChainedOp::CmpOp(a) => BinaryOp::Inequality(a), ChainedOp::MultiEq => BinaryOp::Eq(Mode::Spec), };` so that it explicitly covers all the cases in `ChainedOp`.